### ISSUE TITLE
Installing camlp4 failed; cannot find camlp4orf.

### ISSUE THREAD

**Issue by hendrikboom3** [ocaml/opam-repository#5280]
I originally installed opam from the Debian/Devuan package.
It forced installation of a few others, including ocaml.
Currently ocaml is at 4.01.0-5 and opam is at 1.2.0-1.

Trying to install camlp4 gives me:

```
hendrik@notlookedfor:~/dv/ocaml/learngtk$ opam install camlp4
The following actions will be performed:
 - install   camlp4.4.01+system
=== 1 to install ===

=-=- Synchronizing package archives -=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=

=-=- Installing packages =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=
Copying ~/.opam/repo/default/packages/camlp4/camlp4.4.01+system/files/install to ~/.opam/system/build/camlp4.4.01+system/
Copying ~/.opam/repo/default/packages/camlp4/camlp4.4.01+system/files/check-camlp4.sh to ~/.opam/system/build/camlp4.4.01+system/
Building camlp4.4.01+system:
  sh ./check-camlp4.sh
[ERROR] The compilation of camlp4.4.01+system failed.
Removing camlp4.4.01+system.
  Nothing to do.
[WARNING] Directory /home/hendrik/.opam/system/lib/camlp4 is not empty, not removing


#=== ERROR while installing camlp4.4.01+system ================================#
# opam-version 1.2.0
# os           linux
# command      sh ./check-camlp4.sh
# path         /home/hendrik/.opam/system/build/camlp4.4.01+system
# compiler     system (4.01.0)
# exit-code    1
# env-file     /home/hendrik/.opam/system/build/camlp4.4.01+system/camlp4-13786-e7966f.env
# stdout-file  /home/hendrik/.opam/system/build/camlp4.4.01+system/camlp4-13786-e7966f.out
# stderr-file  /home/hendrik/.opam/system/build/camlp4.4.01+system/camlp4-13786-e7966f.err
### stdout ###
# ...[truncated]
# `opam switch`.
# 
# Here are some installation instructions for camlp4 if you obtained OCaml
# via the OPAM binary packages:
# 
# http://software.opensuse.org/download.html?project=home%3Aocaml&package=ocaml
# 
# * Debian/Ubuntu: sudo apt-get install camlp4-extra
# * RHEL/CentOS/Fedora: sudo yum install ocaml-camlp4
# 
### stderr ###
# ./check-camlp4.sh: 3: ./check-camlp4.sh: camlp4orf: not found

hendrik@notlookedfor:~/dv/ocaml/learngtk$ 
```

-- hendrik

**Comment by hendrikboom3** [comment]
Anyone know why the output in  the issue report s so terribly formatted?  I'd be happy to edit it if I knew what went wrong.

**Comment by hendrikboom3** [comment]
Changing versions, (i.e. opam switch install 4.02.3) and then using the new switch worked wonders.
But it  is still a problem for people using Debian stable without the hand-holding I got on the beginner's mailing list about this problem.

**Comment by OCamlPro-Henry** [comment]
Camlp4 being distributed with the compiler until ocaml-4.01, it was custom when using the system switch to use a "system" `camlp4`. Here, there is a custom error message that suggest you to install `camlp4` from Debian:
```
sudo apt-get install camlp4-extra
```
Is the error message not visible enough ? Otherwise, we could close this bug report.

PS: for the garbled error message on GitHub, you might use verbatim block, see the "Code" tab in: 
 
  https://guides.github.com/features/mastering-markdown/#examples

**Comment by hendrikboom3** [comment]
I remember (probably incorrectly) that I had obtained ocaml by installing the Debian opam package together with its dependencies.  I'm not sure how I got ocaml itself this way, because ocaml is only a recommended package.  Presumbly I saw the recommendations and decided to install it, too, because I knew I needed it.  But camlp4-extra does not seem to be on the list of recommendations, either.  Perhaps this should be corrected in the Debian package.

I indeed did miss the message to install camlp4-extra, and I'm not sure how.  Perhaps because the instruction was prefaced with the clause "if you obtained OCaml via the OPAM binary packages" and Ii had installed it from Debian packages instead.  Perhaps these were actually the very packages you inteded?

Thanks for the markdown link.  I've reformatted the code block.

-- hendrik
